cdap-ui/app/hydrator/services/create/stores/config-store.js

```diff
@@ -413,23 +416,38 @@ class HydratorPlusPlusConfigStore {
     return this.getConfig().properties;
   }
   setProperties(properties) {
+    let numExecutorKey = 'system.spark.spark.executor.instances';
+    let numExecutorOldKey = 'system.spark.spark.master';
+    let backPressureKey = 'system.spark.spark.streaming.backpressure.enabled';
     if (typeof properties !== 'undefined' && Object.keys(properties).length > 0) {
       this.state.config.properties = properties;
     } else {
       this.state.config.properties = {};
-      if (this.state.artifact.name === this.GLOBALS.etlDataStreams) {
-        this.state.config.properties['system.spark.spark.streaming.backpressure.enabled'] = true;
-        if (this.isDistributed) {
-          this.state.config.properties['system.spark.spark.executor.instances'] = 1;
-        } else {
-          this.state.config.properties['system.spark.spark.master'] = 'local[1]';
-        }
+    }
+    if (this.state.artifact.name === this.GLOBALS.etlDataStreams) {
+      if (typeof this.state.config.properties[backPressureKey] === 'undefined') {
+        this.state.config.properties[backPressureKey] = true;
+      }
+    }
+    if (this.getEngine() === 'spark' || this.state.artifact.name === this.GLOBALS.etlDataStreams) {
+      this.state.config.properties[numExecutorKey] = this.state.config.properties[numExecutorKey] || 1;
+      if (this.state.config.properties.hasOwnProperty(numExecutorOldKey)) {
+        let formattedNum = this.state.config.properties[numExecutorOldKey];
+        formattedNum = typeof formattedNum === 'string' ? formattedNum.substring(6, formattedNum.length - 1) : formattedNum;
```

Comment:
Maybe add a comment on where 6 comes from. Also, if the format is string, don't we need to do a parseInt after slicing the string?